crates/bevy_hierarchy/src/components/children.rs

```diff
@@ -152,6 +139,17 @@ impl Deref for Children {
     }
 }
 
+impl<'a> IntoIterator for &'a mut Children {
```

Comment:
I afraid that this will allow mutation which could break the hierarchy...
Maybe we need 2 traits for iteration after all :(

Reply:
Is the situation really all that different from how we previously had a `MapEntities` impl able to mutate children? This is just a little more flexible, which I'll grant might make it a bit more attractive for misuse.

Reply:
Ah, right... But yes, it's a bit more attractive :) Users will be able to mutable iterate over them in a loop.